src/drop.test-d.ts

```diff
@@ -0,0 +1,58 @@
+import { drop } from "./drop";
+import { pipe } from "./pipe";
+
+describe("data first", () => {
+  it("works on regular inputs", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], 2)).toEqualTypeOf<
+      [number, number, number]
+    >();
+  });
+
+  it("works on empty arrays", () => {
+    expectTypeOf(drop([], 2)).toEqualTypeOf<[]>();
+  });
+
+  it("works with negative numbers", () => {
+    expectTypeOf(drop([1, 2, 3], -1)).toEqualTypeOf<[number, number, number]>();
+  });
+
+  it("works when dropping more than the length of the array", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], 10)).toEqualTypeOf<[]>();
+  });
+
+  it("works with generalized typed arrays", () => {
+    expectTypeOf(drop([] as Array<number | string>, 2)).toEqualTypeOf<
+      Array<number | string>
+    >();
+  });
+
+  it("works with generalized typed count", () => {
+    expectTypeOf(drop([1, "2", true], 2 as number)).toEqualTypeOf<
+      Array<boolean | number | string>
+    >();
+  });
+});
```

Comment:
please also add a test for "prefix" tuples
```suggestion
  it("works with prefix arrays", () => {
    const result = drop(["hello", "world"] as [string, string, ...Array<string>], 1);
    expectTypeOf(result).toEqualTypeOf<[string, ...Array<string>]>();
  });
});
```

Reply:
Done